#include<potbot/PathPlanning.h>

void PathPlanningClass::setLaunchParam(){
    
    ros::NodeHandle n("~");
    //n.getParam("",);
    n.getParam("PATH_PLANNING_METHOD",PATH_PLANNING_METHOD);
    n.getParam("PATH_PLANNING_FILE",PATH_PLANNING_FILE);
    n.getParam("USE_AMCL",USE_AMCL);
    n.getParam("PID_CONTROL",PID_CONTROL);
    n.getParam("ANGLE_CORRECTION",ANGLE_CORRECTION);
    n.getParam("MAX_VELOCITY",MAX_VELOCITY);
    n.getParam("MAX_ANGULAR",MAX_ANGULAR);
    n.getParam("TARGET_POSITION_X",TARGET_POSITION_X);
    n.getParam("TARGET_POSITION_Y",TARGET_POSITION_Y);
    n.getParam("TARGET_POSITION_MARGIN",TARGET_POSITION_MARGIN);
    n.getParam("GAIN_PROPORTIONAL",GAIN_PROPORTIONAL);
    n.getParam("GAIN_INTEGRAL",GAIN_INTEGRAL);
    n.getParam("GAIN_DIFFERENTIAL",GAIN_DIFFERENTIAL);
    n.getParam("AVOIDANCE_RADIUS",rho_zero);
    n.getParam("PATH_CREATE_PERIOD",PATH_CREATE_PERIOD);
    n.getParam("POTENTIAL_FIELD_WIDTH",POTENTIAL_FIELD_WIDTH);
    n.getParam("POTENTIAL_FIELD_DIVDE_X",POTENTIAL_FIELD_DIVDE_X);
    n.getParam("POTENTIAL_FIELD_DIVDE_Y",POTENTIAL_FIELD_DIVDE_Y);
    n.getParam("CALCULATE_BIAS",CALCULATE_BIAS);
    n.getParam("POTENTIAL_BIAS_COEFFICIENT_0",POTENTIAL_BIAS_COEFFICIENT_0);
    n.getParam("POTENTIAL_BIAS_COEFFICIENT_1",POTENTIAL_BIAS_COEFFICIENT_1);
    n.getParam("AHEAD_PATH",AHEAD_PATH);
    n.getParam("EXCLUDE_LRF",EXCLUDE_LRF);
}